ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -699,5 +773,31 @@ public String toString() {
           + ", routingNodeType=" + routingNodeType
           + '}';
     }
+
+    public String variantName() {
+      return sourceTypeName() + "-"
+          + planTypeName() + "-"
+          + routingNodeTypeName();
+    }
+
+    public String sourceTypeName() {
+      return getName(sourceType);
+    }
+
+    public String planTypeName() {
+      return getName(planType);
+    }
+
+    public String routingNodeTypeName() {
+      return getName(routingNodeType);
+    }
+
+    private String getName(final Enum<?> o) {
+      if (o == null) {
+        return "unknown";
+      } else {
+        return o.name().toLowerCase();
+      }
+    }
```

Comment:
Consolidated the logic for getting the metric-friendly version of the enum values. This inlines the logic for unknown values, which was previously split between the enum classes, which each had an "UNKNOWN" value, and the metric logic, which would take the lowercase version of the word as the metric dimension.